app/models/concerns/isolation_symptom_onset_validator.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+# Validates that when in Isolation, either:
+# - symptom_onset is set to a value
+# - the patient has at least one positive lab result
+class IsolationSymptomOnsetValidator < ActiveModel::Validator
+  include ValidationHelper
+  def validate(record)
+    return unless record.isolation &&
+                  record.symptom_onset.nil? &&
+                  record.laboratories.where(result: 'positive').where.not(specimen_collection: [nil, '']).length.zero? &&
```

Comment:
Thanks for the feedback, this made me think more deeply about this validator, and I actually think there should be no need to do any query's on the laboratories. Really this validator should _only_ run on `create` of a Patient in the `:api` context. As it is right now, we run it on `update` in the `:api` context, but that isn't right, because then someone could update a lab result, making the corresponding patient invalid, and then go to update that patient and be confused as to why that patient is now invalid. Also in the UI we are going to do soft warnings on those sorts of updates, so it isn't right to run this validator when a patient is updated via the API.

So if I change this to only validate on create, then we shouldn't ever need to query, since the lab results will be in memory and have not been saved to the database yet. Is what I'm saying here making sense?

Reply:
Yeah, based on our discussion today, it sounds like this logic is covid specific and we're only interested in enforcing it on `patient#create`

Reply:
Ok I've updated it so that it only runs on API Create in 50924a9.